### PR TITLE
utils for pdf prints in the browser

### PR DESCRIPTION
**Pull request by stazrad**
Resolves: attach_gh_issue_here

### PR Safety Checklist:

 - [x] Added the task to the appropriate release doc under **Enhancements** or **Bug Fixes**
 - [x] Bump `package.json` & `package-lock.json` version numbers to appropriate release
 - [x] (optional) All external API changes have been documented
 - [x] (optional) Build docs: `npm run docs`

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -3,17 +3,18 @@ export * from './ContextMenu'
...
-export * from './TabbedPanel'
 export * from './LayerStyler'
 export * from './Map'
+export * from './MultiMapManager'
+export * from './Pdf'
 export * from './Popup'
+export * from './Project'
 export * from './Provider'
+export * from './Snackbar'
+export * from './TabbedPanel'
```
Comment (stazrad) [inline src/index.js:18]: alphabetize 👌

File: src/Pdf/utils.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        // convert mm to pixel here
+        const [milimeters] = attribute.split('mm')
+        const pixels = milimeters * 3.78 // mm to pixel conversion
+        
+        return pixels
+      }
+      // add more catches to convert other units here
+
+      return attribute
+    }
```
Comment (stazrad) [inline src/Pdf/utils.js:19]: move to utils

File: src/Pdf/utils.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      height,
+      width,
+    }
+
+    return config
+  }).filter(element => element)
+
+  const template = {
+    elements,
+    dimensions: [793, 1112],
```
Comment (stazrad) [inline src/Pdf/utils.js:44]: get the real dimensions off the svg

File: src/Pdf/utils.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    orientation: 'landscape',
+    fileName: 'kill_geoprint.pdf',
```
Comment (stazrad) [inline src/Pdf/utils.js:46]: calculate orientation & allow name to be passed as a input

File: src/Pdf/utils.js (hunk 2 above)
Comment (stazrad) [reply src/Pdf/utils.js:19]: delete this and just use mm

File: src/Pdf/utils.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  return template
+}
+
+export async function printPDF (template) {
+  const {
+    dimensions,
+    elements,
+    fileName,
+    orientation,
+  } = template
```
Comment (stazrad) [inline src/Pdf/utils.js:58]: add unit to the template

File: src/Pdf/utils.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+      y,
+      height,
+      width,
+    } = element
+
+    console.log('template element', element)
+
+    if (type === 'image') {
+      // http://raw.githack.com/MrRio/jsPDF/master/docs/module-addImage.html
+      doc.addImage(content, 'JPEG', x, y, width, height, id, 'NONE', 0)
```
Comment (stazrad) [inline src/Pdf/utils.js:80]: add a uri option for the template input

File: src/Pdf/utils.js [hunk 7]
```diff
@@ -1,31 +1,45 @@
...
+  else if (svgHeight.split('in').length > 1) unit = 'in'
+  const numberify = string => {
+    const [numberAsString] = string.split(unit) // closure, order matters
+
+    return Number(numberAsString)
+  }
+  // calc dimensions from svg
+  const height = numberify(svgHeight)
+  const width = numberify(svgWidth)
+  const orientation = width > height ? 'landscape' : 'portrait'
```
Comment (IDisposable) [inline src/Pdf/utils.js:29]: What if (not likely), the width and height are in different units? Are we going to break this orientation computation?

File: src/Pdf/utils.js [hunk 8]
```diff
@@ -1,31 +1,45 @@
...
     const element = svgDoc.getElementById(id)
     if (!element) {
       ugh.warn(`Element with id '${id}' is missing from svg document`)
       return undefined // this will be filtered out
     }
-    const x = convertToPixels(element.getAttribute('x'))
-    const y = convertToPixels(element.getAttribute('y'))
-    const height = convertToPixels(element.getAttribute('height'))
-    const width = convertToPixels(element.getAttribute('width'))
+    const x = numberify(element.getAttribute('x'))
```
Comment (IDisposable) [inline src/Pdf/utils.js:39]: Should we normalize these if mixes units?

File: src/Draw/__snapshots__/DrawContainer.test.js.snap [hunk 9]
```diff
@@ -164,5 +165,5 @@ exports[`<DrawContainer /> should render a basic prebuilt DrawContainer componen
               [36m</span>[39m
             [36m</div>[39m
             [36m<div[39m
-              [33mclass[39m=[32m\\"MuiInputBase-root MuiInput-root MuiI..."
+              [33mclass..."
```
Comment (IDisposable) [inline src/Draw/__snapshots__/DrawContainer.test.js.snap:168]: Seems like an unrelated change, worthy of a justification comment.

File: src/Pdf/utils.js (hunk 7 above)
Comment (stazrad) [reply src/Pdf/utils.js:29]: that would be pretty edgy but yea i could see a `110mm` height with a `100%` width that may break this?

File: src/Pdf/utils.js (hunk 8 above)
Comment (stazrad) [reply src/Pdf/utils.js:39]: like convert them all to pixels or something?

File: src/Draw/__snapshots__/DrawContainer.test.js.snap (hunk 9 above)
Comment (stazrad) [reply src/Draw/__snapshots__/DrawContainer.test.js.snap:168]: snapshot updated from this: https://github.com/Bayer-Group/ol-kit/pull/309/files/99ad173cd0eafa255daa035de3bdc138b85dea74#diff-361b8ea1a952277c1efc12a7e05d9fae101b9480f46587dd4019ba19c03fc615R25

(but can be reverted)

File: src/Pdf/utils.js (hunk 8 above)
Comment (IDisposable) [reply src/Pdf/utils.js:39]: Not sure `px` would be the safest as that's mapped from the realm DPI... but wondering if just whining about mixed units would be sufficient? If not, probably target `px` if any of them are `['px','pt']` ... and `mm` if any other unit... and what to do with `'%'`???

File: src/Pdf/utils.js (hunk 7 above)
Comment (IDisposable) [reply src/Pdf/utils.js:29]: Agreed... just worried that it wouldn't be obvious why it was behaving oddly. Perhaps merely asserting that the units are the same it good enough?

File: src/Draw/__snapshots__/DrawContainer.test.js.snap (hunk 9 above)
Comment (IDisposable) [reply src/Draw/__snapshots__/DrawContainer.test.js.snap:168]: Perfect. Helps to point out this is a generated snap. Thanks.

**Review by AtherBilal — APPROVED** [review]
